File: 2_december/framingnames.js
function createFrame(names) {
    const maxLength = Math.max(...names.map(name => name.length));
    const border = '*'.repeat(maxLength + 4);

    const frame = names.map(
        name => `* ${name.padEnd(maxLength)} *`
    ).join('\n');

    return `${border}\n${frame}\n${border}`;
}

console.log(createFrame(['midu', 'madeval', 'educalvolpz']))

File: 2_december/framingnames.py
def createFrame(names):
  max_length = max(map(len, names))
  long_text = filter(lambda x: len(x) == max_length, names)
  lentgh = len(list(long_text)[0])+4
  border = '*' * lentgh
  frame = border+'\n'+'\n'.join(map(lambda x: '* '+x.ljust(max_length)+' *', names))+'\n'+border
  return frame

names = ['midu', 'madeval', 'educalvolpz']
print(createFrame(names))